src/test/TEST_ij/amgdd.saved.florentia

```diff
@@ -1,64 +1,64 @@
 # Output file: amgdd.out.900
 BoomerAMG-DD Iterations = 9
-Final Relative Residual Norm = 8.292318e-09
+Final Relative Residual Norm = 7.722167e-09
 
 # Output file: amgdd.out.901
 BoomerAMG-DD Iterations = 10
-Final Relative Residual Norm = 6.428957e-09
+Final Relative Residual Norm = 6.234339e-09
 
 # Output file: amgdd.out.902
 BoomerAMG-DD Iterations = 10
-Final Relative Residual Norm = 5.654168e-09
+Final Relative Residual Norm = 5.084501e-09
 
 # Output file: amgdd.out.903
 BoomerAMG-DD Iterations = 23
-Final Relative Residual Norm = 9.249401e-09
+Final Relative Residual Norm = 9.617543e-09
 
 # Output file: amgdd.out.904
 BoomerAMG-DD Iterations = 23
-Final Relative Residual Norm = 9.249401e-09
+Final Relative Residual Norm = 9.617543e-09
 
 # Output file: amgdd.out.905
 BoomerAMG-DD Iterations = 10
-Final Relative Residual Norm = 4.748844e-09
+Final Relative Residual Norm = 4.879019e-09
 
 # Output file: amgdd.out.906
 BoomerAMG-DD Iterations = 10
-Final Relative Residual Norm = 4.607814e-09
+Final Relative Residual Norm = 4.895847e-09
 
 # Output file: amgdd.out.910
 GMRES Iterations = 7
-Final GMRES Relative Residual Norm = 9.441124e-10
+Final GMRES Relative Residual Norm = 8.874321e-10
 
 # Output file: amgdd.out.911
 GMRES Iterations = 7
-Final GMRES Relative Residual Norm = 5.299116e-09
+Final GMRES Relative Residual Norm = 3.507660e-09
 
 # Output file: amgdd.out.912
 GMRES Iterations = 7
-Final GMRES Relative Residual Norm = 3.801730e-09
+Final GMRES Relative Residual Norm = 4.268209e-09
 
 # Output file: amgdd.out.913
 GMRES Iterations = 6
-Final GMRES Relative Residual Norm = 1.613363e-09
+Final GMRES Relative Residual Norm = 1.690129e-09
 
 # Output file: amgdd.out.914
 GMRES Iterations = 15
-Final GMRES Relative Residual Norm = 3.727440e-09
+Final GMRES Relative Residual Norm = 3.193093e-09
 
 # Output file: amgdd.out.915
 GMRES Iterations = 5
-Final GMRES Relative Residual Norm = 7.111268e-09
+Final GMRES Relative Residual Norm = 7.894937e-09
 
 # Output file: amgdd.out.916
-GMRES Iterations = 21
-Final GMRES Relative Residual Norm = 1.801927e-03
+GMRES Iterations = 15
```

Comment:
Yes, I think this is fine. I tested the cuda build on lassen and saw similarly significant differences when switching spgemm implementations (our custom vs. cusparse in that case). I was a bit surprised there were any differences at all when switching spgemm implementations, but there is a good explanation. Different implementations give different column index orderings, and this can cause slightly different behavior down the line for some pieces of AMG, notably coarsening. I've observed slightly different AMG hierarchies generated when doing the same run and just switching spgemm implementations. Note that AMG-DD convergence is particularly sensitive to the selection of coarse grids, thus the large difference here.

Reply:
Sounds good.  Thanks!